### PR TITLE
feat: ability to fetch spaces convos

### PR DESCRIPTION
**Pull request by ashgansh**
## What does this PR do?

This PR introduces a new feature that allows viewing and managing conversations within specific spaces. It includes the following main changes:

1. Adds a new `getSpaceConvos` query to the `spaceRouter` to fetch conversations for a specific space.
2. Creates new components and routes for displaying space-specific conversations.
3. Refactors the existing conversation list components to be more flexible and reusable between org-wide and space-specific contexts.
4. Updates the sidebar navigation to link to the new space-specific conversation views.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Required

- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [x] Commented on my code in hard-to-understand areas
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

File: apps/web/src/app/[orgShortcode]/convo/_components/org-convo-list.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      orgShortcode,
+      includeHidden: props.hidden ? true : undefined
+    },
+    {
+      getNextPageParam: (lastPage) => lastPage.cursor ?? undefined,
+      staleTime: ms('1 hour')
+    }
+  );
+  if (!convos) return null;
+  const allConvos = convos.pages.flatMap((page) => page.data);
```
Comment (BlankParticle) [inline apps/web/src/app/[orgShortcode]/convo/_components/org-convo-list.tsx:33]: wrap this in a `useMemo`

File: apps/web/src/app/[orgShortcode]/convo/_components/org-convo-list.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      convos={allConvos}
+      isLoading={isLoading}
+      hasNextPage={hasNextPage}
+      isFetchingNextPage={isFetchingNextPage}
+      fetchNextPage={fetchNextPage}
+      ConvoItem={(itemProps) => (
+        <ConvoItem
+          {...itemProps}
+          link={`/${orgShortcode}/convo/${itemProps.convo.publicId}`}
+        />
```
Comment (BlankParticle) [inline apps/web/src/app/[orgShortcode]/convo/_components/org-convo-list.tsx:47]: extract `ConvoItem`, wrap in `useCallback` make it like so you can pass
```ts
ConvoItem={ConvoItemRenderer}
```

File: apps/web/src/app/[orgShortcode]/convo/layout.tsx [hunk 3]
```diff
@@ -318,23 +324,40 @@ export default function Layout({
...
           <ChildrenWithOrgIssues>{children}</ChildrenWithOrgIssues>
         ))}
     </div>
   );
 }
+export default function Layout({ children }: { children: React.ReactNode }) {
+  const [showHidden, setShowHidden] = useState(false);
+  return (
+    <ConvoLayoutWrapper
+      convoList={<OrgConvoList hidden={showHidden} />}
```
Comment (BlankParticle) [inline apps/web/src/app/[orgShortcode]/convo/layout.tsx:357]: define `OrgConvoList` outside the return block, wrap in a `useMemo`/`useCallback` pass that

**Review by BlankParticle — CHANGES_REQUESTED** [review]

**Comment by McPizza0** [comment]
we will fix the issues in a separate PR

**Review by McPizza0 — APPROVED** [review]

**Pull request merged**